### PR TITLE
SL-1736: Search

### PR DESCRIPTION
**Pull request by chohmann**
- Search table migration
- updated discovery migration to not use enum types and to have data and data_hash columns on node_version_history instead of node_version table

### CODE REVIEW

File: db/migrate/20190116182055_add_discovery_tables.rb [hunk 1]
```diff
@@ -8,80 +8,11 @@ class AddDiscoveryTables < ActiveRecord::Migration
   def up
     if Gitlab::Database.postgresql?
       execute <<-EOF
```
Comment (casserni) [inline db/migrate/20190116182055_add_discovery_tables.rb]: super minor can you update `EOF` to `SQL`, using SQL gives use syntax highlighting in vscode

**Comment by collinbachi** [comment]
Is this expected to fix the migration issue when starting gitlab via dev-kit? + @casserni

**Comment by chohmann** [comment]
Yes, this should fix the issues the custom enum types in Postgres were causing.

**Review by pytlesk4 — APPROVED** [review]
👶

**Pull request merged**